api_tests/harness.js

```diff
@@ -16,202 +20,267 @@ const project_root = execSync("git rev-parse --show-toplevel", {
+                comit_nodes.map(([name, _]) => name).join(", ")
+        );
+        for (let [name, comit_config] of comit_nodes) {
+            if (this.running_nodes[name]) {
+                continue;
+            }
 
-async function startBtsieve(name, btsieve_config) {
-    return await spawn(project_root + "/target/debug/btsieve", [], {
-        cwd: services_cwd,
-        encoding: "utf-8",
-        env: btsieve_config.env,
-        stdio: [
-            "ignore",
-            fs.openSync(log_dir + "/btsieve-" + name + ".log", "w"),
-            fs.openSync(log_dir + "/btsieve-" + name + ".log", "w"),
-        ],
-    });
-}
+            this.running_nodes[name] = await spawn(
+                project_root + "/target/debug/comit_node",
+                [],
+                {
+                    cwd: services_cwd,
+                    encoding: "utf-8",
+                    env: { COMIT_NODE_CONFIG_PATH: comit_config.config_dir },
+                    stdio: [
+                        "ignore",
+                        fs.openSync(
+                            log_dir + "/comit_node-" + name + ".log",
+                            "w"
+                        ),
+                        fs.openSync(
+                            log_dir + "/comit_node-" + name + ".log",
+                            "w"
+                        ),
+                    ],
+                }
+            );
+        }
+
+        await util.sleep(500);
+    }
 
-// ********************************** //
-// Start services, run test, shutdown //
-// ********************************** //
-
-function run_tests(file) {
-    describe("🏃" + file, async function() {
-        let block_interval;
-        let subprocesses = [];
-        before(async function() {
-            this.timeout(ledger_up_time + 4000);
-
-            if (config.ledgers) {
-                const subprocess = await startDockerContainers(
-                    config.ledgers,
-                    ledgers_config
-                );
-                subprocesses.push(subprocess);
-                await util.sleep(ledger_up_time);
+    stop_comit_nodes() {
+        let names = Object.keys(this.running_nodes);
+
+        if (names.length > 0) {
+            console.log("Stopping comit nodes: " + names.join(", "));
+            for (let [_, process] of Object.entries(this.running_nodes)) {
+                process.kill();
             }
+            this.running_nodes = {};
+        }
+    }
+}
+
+class BtsieveRunner {
+    constructor() {
+        this.running_btsieves = {};
+    }
 
-            if (config.btsieve) {
-                Object.keys(config.btsieve).forEach(async function(name) {
-                    const subprocess = await startBtsieve(
-                        name,
-                        config.btsieve[name]
-                    );
-                    subprocesses.push(subprocess);
-                });
+    async ensure_btsieve_running(btseives) {
```

Comment:
:x: The word "btseive" is used in several places instead of the correct "btsieve".